### PR TITLE
Shift multi select

### PR DESCRIPTION
**Pull request by raheelx**
Hi @JoshEngebretson :) 

This commit fixes the issue we spoke about on Issue #376 - does not select children objects if parent is not expanded. I also changed the way shift selecting happened to make it select around a pivot instead.

Please let me know if there is anything else you'd like changed or added. 

-Raheel

### CODE REVIEW

File: Source/Atomic/UI/UIListView.h [hunk 1]
```diff
@@ -98,6 +98,10 @@ class UIListView : public UIWidget
 
     void SelectItem(ListViewItem* item, bool select);
 
+    ListViewItem* pivot;
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UIListView.h]: member variables should be post-pended with an underscore, so pivot_, pivotIndex_, startNewSelection_

File: Source/Atomic/UI/UIListView.cpp [hunk 2]
```diff
@@ -938,11 +945,19 @@ bool UIListView::OnEvent(const tb::TBWidgetEvent &ev)
...
+                    startNewSelection = false;
                 }
                 else
                 {
                     SelectSingleItem(item, false);
+
+                    pivot = item;
+                    pivotIndex = i;
+                    startNewSelection = true;
                 }
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UIListView.cpp]: This all looks great, nice cleanups with additional functionality 👍

**Comment by mattbenic** [comment]
Hi @JoshEngebretson, do you need any changes to this before it can be landed?

**Comment by JoshEngebretson** [comment]
@mattbenic @raheelx Yes, commented above about member variable naming convention

**Comment by mattbenic** [comment]
Oh bloody hell, missed that one sorry.

**Comment by JoshEngebretson** [comment]
No problem, I usually try and do a little summary post at the end of a PR mentioning if there is an issue, easy to miss!

**Comment by JoshEngebretson** [comment]
@AtomicBuildBot add to whitelist, test this please

**Comment by JoshEngebretson** [comment]
@AtomicBuildBot retest this please

**Comment by raheelx** [comment]
Hi @JoshEngebretson :) sorry this took so long! I just pushed up the changes you wanted

**Comment by JoshEngebretson** [comment]
@raheelx Awesome, thanks for the updates, 9 successful checks!  LANDING! :)

**Pull request merged**